File: lib/models/item.dart
import 'package:decimal/decimal.dart';
import 'package:flutter_application_1/models/table_item.dart';

class Item extends TableItem {
  final String itemCode;
  final String itemName;
  final String type;
  final String series;
  final Decimal price;

  Item(
      {required this.itemCode,
      required this.itemName,
      required this.type,
      required this.series,
      required this.price});

  factory Item.fromJson(Map<String, dynamic> json) {
    return Item(
      itemCode: json['item_code'],
      itemName: json['item_name'],
      type: json['item_type'],
      series: json['series'],
      price: Decimal.parse(json['price']),
    );
  }
  @override
  Map<String, dynamic> toTableData() {
    return {
      'itemCode': itemCode,
      'itemName': itemName,
      'type': type,
      'series': series,
      'price': price
    };
  }

  static const columnsForLargeScreen = [
    'itemCode',
    'itemName',
    'type',
    'series',
    'price',
  ];

  static const columnsForMediumScreen = [
    'itemCode',
    'itemName',
    'type',
    'series',
    'price',
  ];

  static const columnsForSmallScreen = [
    'itemCode',
    'itemName',
    'price',
  ];
}

File: srcipts/item.py
from sqlalchemy import Column, DECIMAL, String
from base import Base


class Item(Base):
    __tablename__ = "item"
    item_id = Column(String(100), primary_key=True, nullable=False)
    item_code = Column(String(255), nullable=False)
    item_name = Column(String(255), default=None)
    item_type = Column(String(255), default=None)
    series = Column(String(255), default=None)
    price = Column(DECIMAL(10, 0), default=None)
